Expand Interfaces for Path Awareness (#30)
scionassociation/scion-research_I-D: draft-meynell-panrg-scion-research-questions.md

* What time synchronization precision should we expect at the border router level of every AS? How far can we go realistically?
-->

# Interfacing SCION with existing technologies
The questions posed here are:

* Are there any existing protocols/solutions that could be adapted to carry SCION traffic?

> * If there are not suitable out-of-the-box, what would be required to make them work?

Reviewer: Sorry, Github refuses to do replacement suggestion because there is a deleted line in there...

```
# Interfacing SCION with existing technologies
The questions posed are about adapting SCION to be interoperable or compatible with existing technologies and protocols:
* For which existing technologies/protocols do we want compatibility/interoperability?
* How would we need to adapt SCION?
* How much work is it?
```

Author: @knmeynell Can you PTAL at this section and write it in proper english?
We tried to be clear, but we both don't agree on the exact words to use here.
The idea is explained in the text of this section. Please feel free to ask us about what we really want to express here :smile: (maybe we don't exactly know what we want :wink: )

Reviewer 2: JFYI Kevin is off this week, he might not respond

-->

# Interfaces for Path Awareness

> The questions posed here are:

Reviewer: If we refer to https://www.rfc-editor.org/rfc/rfc9217.html, I think `Interfaces for Path Awareness`  are something different than what is discussed here. I think they mean interfaces for applications/users to select paths.

However, this chapter discusses more the interfacing of SCION with existing internet protocols/solution.

Maybe rename this section to `Interfacing SCION with existing technologies`?

> * Are there any existing protocols/solutions that can carry path aware traffic?

Reviewer: Maybe we should be a bit more precise, I think there are a lot of solution for path aware traffic. How about:
`- Are there any existing protocols/solution that can be adapted to carry SCION traffic?`
?

> * If there are not, what would it require?

Reviewer: In the next section you show that there are existing protocols/solutions, so the question is already answered?
Remove this?

Author: I think there aren't. The list is there because ISPs will mention IPv6, and vendors will say that it is not really a solution. Same for the scion-ip translation approach.
I've modified the sentence to `If there are not suitable out-of-the-box, what would it require to make them work?`

Reviewer: Okay, so I think I misunderstood what this is about. Maybe we should be more clear what this question is about?

> * Are there any existing protocols/solutions that could be adapted to carry SCION traffic?
>  * If there are not suitable out-of-the-box, what would be required to make them work?

I think the question isn't really whether something could be adapted, but whether it would be beneficial if something would be adapted.

I think the question may be more like: 
```
* What existing protocols/solutions do we need to be interoperable/compatible with SCION?
* If they are not interoperable/compatible with SCION yet, how can this be achieved?
```
Does that make sense?

Author: The question is about how to make SCION interoperable with existing tech. such as IPv6. And, if not out-of-the-box, what is the effort required for that.
It's more about reusing something that exists to carry SCION traffic, that enabling SCION for existing things. Essentially: what is the least resistance path to allow ISPs to carry SCION traffic?
I like your second sentence better than my original, I propose something like:
```
* Are there any existing protocols/solutions that could carry SCION traffic?
* If they are not interoperable/compatible with SCION yet, how can this be achieved?
```
What do you think?

* What time synchronization precision should we expect at the border router level of every AS? How far can we go realistically?
-->

# Interfacing SCION with Existing Technologies
The questions posed here are:

* What existing protocols/solutions should be compatible with SCION simultanously?

>   Referring to the probable fact that ISPs want to offer traditional IP services aside from SCION.

Reviewer 2: ```suggestion
  How can ISPs offer traditional IP side by side with SCION.
```

> * What existing technologies can replace part of the SCION specification?
>   Referring to the possibility of slightly changing an existing technology (e.g. IPv6) to be used
>   as part of SCION, replacing part of the ad-hoc specification that we have for SCION.

Reviewer 2: ```suggestion
* Could a future evolution of the SCION specification better reuse existing technologies?
  Referring to the possibility of slightly changing an existing technology (e.g. IPv6) to be used
  as part of SCION, replacing part of the ad-hoc specification that we have for SCION.
```

* What would be required effort to make them work?
  Referring to the ranking according to different types of parties involved: ISPs, vendors,

>   application developers, etc.

Reviewer 2: Is this about deployment incentives? Or Implementation effort?

Author: Incentives vs. effort, for the different parties.
